network-mux/src/Network/Mux/DeltaQTraceStatsSupport.hs

```diff
@@ -12,7 +13,14 @@ where
 --
 -- Definite space/time optimisation task here.
 
-import Network.Mux.DeltaQTraceTypes
+import           Network.Mux.DeltaQTraceTypes
 
+import qualified Data.Vector.Unboxed as V
+import           Statistics.LinearRegression
+  
 estimateGS :: [(Int, SISec)] -> (Double, Double, Double)
-estimateGS = undefined
+estimateGS xys
+  = let (xs', ys') = unzip xys
+        xs = V.fromList $ map fromIntegral xs'
+        ys = V.fromList $ map (\(S x) -> fromRational . toRational $ x) ys' 
```

Comment:
Rather than `fromRational . toRational`, use `realToFrac` which is optimised for most conversion pair types.

Reply:
will do